strapi/strapi
# feat(cli): trigger login sequence when token is missing or invalid

### What does it do?

When a user who is not logged in tries to execute a command that requires authentication, the system prompts the user with an option to log in. If the user confirms, the login process is initiated, allowing the user to authenticate before proceeding with the command.

### Why is it needed?

Currently, when a user who is not logged in or has a non-existent, invalid, or expired token tries to execute a command that requires authentication, the process fails, and the command execution is halted.

### How to test it?

Scenario: Using the deploy command without being logged in
- Expected Behavior:
	- The user is informed that they need to log in to use the command and is asked if they want to log in.
	- If the user selects login confirmation, the login flow is launched.
	- After successful authentication, the deploy or project creation flow continues.
	- If the user declines to log in, nothing else happens.

Scenario: Using the deploy command with an expired token
- Expected Behavior:
	- The user is informed that their token has expired and that they need to log in again.
	- The user is asked if they want to log in.
	- If the user selects login confirmation, the login flow is launched.
	- After successful authentication, the deploy or project creation flow continues.
	- If the user declines to log in, nothing else happens.


### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request
